src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -478,8 +478,7 @@ HRESULT STDMETHODCALLTYPE CorProfiler::JITCompilationStarted(
           caller.name, "()");
   }
 
-  if (!ProfilerAssemblyIsLoadedIntoAppDomain(module_metadata->app_domain_id) &&
```

Comment:
Why is `ProfilerAssemblyIsLoadedIntoAppDomain` not needed anymore? Isn't it necessary for .NET Framework?

Reply:
The purpose of `Datadog.Trace.ClrProfiler.Managed.Loader.dll` was to load `Datadog.Trace.ClrProfiler.Managed.dll` early and disable tracing if we can't load it.

The check for `ProfilerAssemblyIsLoadedIntoAppDomain()` is an optimization that skipped loading `Datadog.Trace.ClrProfiler.Managed.Loader.dll` and running the startup hook if we detect that the assembly is already loaded.

I removed that condition because now I want to always load `Datadog.Trace.ClrProfiler.Managed.Loader.dll` and run the startup hook, even if we detect that `Datadog.Trace.ClrProfiler.Managed.dll` is already loaded. There is new code in there used to subscribe to `DiagnosticSource` events.